fractal_server/main.py

```diff
@@ -140,6 +144,32 @@ def start_application() -> FastAPI:
             The fully initialised application.
     """
     app = FastAPI(lifespan=lifespan)
+
+    settings = Inject(get_settings)
+
+    @app.middleware("http")
+    async def slow_response_middleware(request: Request, call_next):
+        logger = set_logger("slow-response")
+        # Measure process time
+        start_timestamp = get_timestamp()
+        start_time = time.perf_counter()
+        response = await call_next(request)
+        stop_time = time.perf_counter()
+        process_time = stop_time - start_time
+        # Log if process time is too high
+        if process_time > settings.FRACTAL_LONG_REQUEST_TIME:
+            end_timestamp = start_timestamp + timedelta(seconds=process_time)
+            warning_message_components = [
+                f"{request.method} {request.url.path}",
+                f"{response.status_code}",
+                f"{process_time:.2f} seconds",
+                f"{start_timestamp.strftime('%Y-%m-%d %H:%M:%S')}",
+                f"{end_timestamp.strftime('%Y-%m-%d %H:%M:%S')}",
+            ]
+            logger.warning(" - ".join(warning_message_components))
```

Comment:
Any special reason for building a list of strings and then joining them, rather than directly building a single string?